tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/ExtensionLibraryViewModel.cs

```diff
@@ -69,6 +71,10 @@ public async Task LoadedAsync()
     {
         await GetInstalledPackagesAndExtensionsAsync();
         GetAvailablePackages();
+
+        // Send a message to the Environments page to let it know that the Extensions page has been loaded.
+        // And that the Environments page should update its list of environments.
+        WeakReferenceMessenger.Default.Send(new ExtensionsPageLoaded(true));

```

Comment:
Hmm, I'm not sure this is the right approach we should use here. I believe it would be more efficient if the `ExtensionService` contained an event that anyone in Dev Home could subscribe to. The payload of the event could then have the provider that was disabled/enabled. The subscribers can then decide what to do with that information. This way we aren't sending messages every time the user enters the page, we'll only send an event when an actual user action takes place. This would also help if we ever add another place in Dev Home to disable a provider outside of the extensions page.


Side note: If we use the events model, and the user disables/enables a provider the environments page shouldn't refresh until the user goes back to the page and only if the provider is a compute system provider. We'd also have to think about what should happen if the page is currently refreshing and the user goes to the extensions page and disables/enables a provider. We'd need to wait for the current one to finish then update the page. I'm in favor of refreshing only when a compute system provider is enabled. Disabling a provider should only remove it from the UI list without using the ComputeSystemService to re-query all the compute system providers with out of proc calls.

Reply:
Agreed -- an enabled/disabled event in the Extension service seems like the right thing to do here. (I'm assuming we're doing this at all because it's too much work to completely reload the Environments page each time we navigate there? We wouldn't want to check for what's enabled and load from whatever is each time we navigate? Which seems the most simple)

Reply:
Changed.